dp-hello-world-event/service/service.go

```diff
@@ -0,0 +1,158 @@
+package service
+
+import (
+	"context"
+	"github.com/ONSdigital/dp-hello-world-event/config"
+	"github.com/ONSdigital/dp-hello-world-event/event"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+// Service contains all the configs, server and clients to run the event handler service
+type Service struct {
+	config        *config.Config
+	server        HTTPServer
+	router        *mux.Router
+	serviceList   *ExternalServiceList
+	healthCheck   HealthChecker
+	consumer      KafkaConsumer
+	eventConsumer EventConsumer
+}
+
+// Run the service
+func Run(ctx context.Context, serviceList *ExternalServiceList, buildTime, gitCommit, version string, svcErrors chan error) (*Service, error) {
+	log.Event(ctx, "running service", log.INFO)
+
+	// Read config
+	cfg, err := config.Get()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to retrieve service configuration")
+	}
+	log.Event(ctx, "got service configuration", log.Data{"config": cfg}, log.INFO)
+
+	// Get HTTP Server with collectionID checkHeader middleware
+	r := mux.NewRouter()
+	s := serviceList.GetHTTPServer(cfg.BindAddr, r)
+
+	// Get Kafka consumer
+	consumer, err := serviceList.GetKafkaConsumer(ctx, cfg)
+	if err != nil {
+		log.Event(ctx, "failed to initialise kafka consumer", log.FATAL, log.Error(err))
+		return nil, err
+	}
+
+	// Event Handler for Kafka Consumer with the created S3 Clients and Vault
+	eventConsumer := event.NewConsumer()
+	eventConsumer.Consume(ctx, consumer, &event.HelloCalledHandler{})
+
+	// Get HealthCheck
+	hc, err := serviceList.GetHealthCheck(cfg, buildTime, gitCommit, version)
+	if err != nil {
+		log.Event(ctx, "could not instantiate healthcheck", log.FATAL, log.Error(err))
+		return nil, err
+	}
+
+	if err := registerCheckers(ctx, hc, consumer); err != nil {
+		return nil, errors.Wrap(err, "unable to register checkers")
+	}
+
+	r.StrictSlash(true).Path("/health").HandlerFunc(hc.Handler)
+	hc.Start(ctx)
+
+	// Run the http server in a new go-routine
+	go func() {
+		if err := s.ListenAndServe(); err != nil {
+			svcErrors <- errors.Wrap(err, "failure in http listen and serve")
+		}
+	}()
+
+	return &Service{
+		config:        cfg,
```

Comment:
The config appears to be stored on the service struct only for accessing `svc.config.GracefulShutdownTimeout`. Is it worth storing only the `GracefulShutdownTimeout` instead of the whole config?